packages/server/src/constraints.ts

```diff
@@ -0,0 +1,114 @@
+import { DAY, HOUR, cborEncode, check } from '@atproto/common'
+import * as plc from '@did-plc/lib'
+import { ServerError } from './error'
+import { parseDidKey } from '@atproto/crypto'
+
+export function assertValidIncomingOp(
+  op: unknown,
+): asserts op is plc.OpOrTombstone {
+  const byteLength = cborEncode(op).byteLength
+  if (byteLength > 4000) {
```

Comment:
my intuition is that 4k might be around the upper end of legit ops, but I don't have any evidence for that so is fine for now. easier to bump larger if there is a need

Reply:
For reference, the ops our server puts out right now are right around 500 bytes. 

Agreed that can bump this later if it turns out to be too tight